src/Zicht/Bundle/AdminBundle/AdminMenu/EventPropagationBuilder.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * @author Boudewijn Schoon <boudewijn@zicht.nl>
+ * @copyright Zicht Online <http://zicht.nl>
+ */
+
+namespace Zicht\Bundle\AdminBundle\AdminMenu;
+
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Zicht\Bundle\AdminBundle\Event\AdminEvents;
+use Zicht\Bundle\AdminBundle\Event\MenuEvent;
+use Zicht\Bundle\AdminBundle\Event\PropagationInterface;
+use Zicht\Bundle\PageBundle\Event\PageViewEvent;
+
+/**
+ * Propagates a PageView event as an AdminMenu event.
+ */
+class EventPropagationBuilder implements PropagationInterface
+{
+    /**
+     * @var array
+     */
+    private $hosts;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    public function __construct(RequestStack $requestStack, array $hosts, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->hosts = $hosts;
+        $this->requestStack = $requestStack;
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    /**
+     * Build the relevant event and forward it.
+     *
+     * @param Event $e
+     * @return void
+     */
+    public function buildAndForwardEvent(Event $e)
+    {
+        if (!$e instanceof PageViewEvent) {
+            return;
+        }
+        /** @var \Zicht\Bundle\PageBundle\Event\PageViewEvent $e */
+
+        /** @var Request $request */
+        if ($request = $this->requestStack->getMasterRequest()) {
+            $host = $request->getHost();
+            foreach ($this->hosts as $otherHost) {
+                if ($otherHost != $host) {
+                    $this->eventDispatcher->dispatch(
+                        AdminEvents::MENU_EVENT,
+                        new MenuEvent(sprintf('%s:\\\\%s%s', $request->getScheme(), $otherHost, $request->getRequestUri()), $otherHost)
```

Comment:
Ziet er allemaal mooi uit. Maar wat doet een MenuEvent? :smile:

Reply:
Via dit event kun je de linkjes linksbovenin beinvloeden die achter het Zichtlogo zitten. Bij artez is er oa behoefte aan een speciaal event dat ook reageert op andere domeinen (vanwege de Finals-constructie https://github.com/zicht/artez.nl/blob/a641cb1605f68b8cf1fb827ab3106d9c68b60041/app/config/bundles/zicht_dev.yml#L3)

Reply:
En ik zag dat dit gekopieerd was uit een PR van Philip uit een andere bundle (waar het niet thuis hoorde), dus :+1: